grails-app/views/dashboard/megamenu.gsp

```diff
@@ -243,7 +252,28 @@
         </g:hasRoleInvoice>
     </g:if>
 
+    <g:if test="${megamenuConfig.stockRequest.enabled && User.get(session?.user?.id)?.getHighestRole(session.warehouse).roleType == RoleType.ROLE_AUTHENTICATED}">
```

Comment:
I don't like the database lookup being done here. Is there a way we could move this to a taglib?

Reply:
Or something like this
```
AuthService.currentUser.get().isPrimaryRoleAtLeast(RoleType.ROLE_AUTHENTICATED)
```
Method naming is not my forte so if someone has a better name please let me know.

Reply:
I would like to check if role is not higher than authenticated. I'm not interested in having here role higher than authenticated.